c7n/resources/s3.py

```diff
@@ -2762,6 +2762,288 @@ def get_inventory_delta(self, client, inventory, b):
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: s3-apply-intelligent-tiering-config
+                resource: aws.s3
+                filters:
+                  - not:
+                    - type: intelligent-tiering
+                      attrs:
+                        - Status: Enabled
+                        - Filter:
+                            And:
+                              Prefix: helloworld
+                              Tags:
+                                - Key: Hello
+                                  Value: World
+                        - Tierings:
+                          - Days: 123
+                            AccessTier: ARCHIVE_ACCESS
+                actions:
+                  - type: set-intelligent-tiering
+                    Id: c7n-default
+                    IntelligentTieringConfiguration:
+                      Id: c7n-default
+                      Status: Enabled
+                      Tierings:
+                        - Days: 149
+                          AccessTier: ARCHIVE_ACCESS
+
+              - name: s3-delete-intelligent-tiering-configuration
+                resource: aws.s3
+                filters:
+                  - type: intelligent-tiering
+                    attrs:
+                      - Status: Enabled
+                      - Id: test-config
+                actions:
+                  - type: set-intelligent-tiering
+                    Id: test-config
+                    Status: delete
+
+              - name: s3-delete-intelligent-tiering-matched-configs
+                resource: aws.s3
+                filters:
+                  - type: intelligent-tiering
+                    attrs:
+                      - Status: Enabled
+                      - Id: test-config
+                actions:
+                  - type: set-intelligent-tiering
+                    Id: matched
+                    Status: delete
+
+    """
+
+    annotation_key = 'c7n:ListItemMatches'
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'oneOf': [
+            {'required': ['type', 'Id', 'IntelligentTieringConfiguration']},
+            {'required': ['type', 'Id', 'Status']}],
+        'properties': {
+            'type': {'enum': ['set-intelligent-tiering']},
+            'Id': {'type': 'string'},
+            # delete intelligent tier configurations via status: delete
+            'Status': {'type': 'string', 'enum': ['delete']},
+            'IntelligentTieringConfiguration': {
+                'type': 'object',
+                'items': {
+                    'type': 'object',
+                    'required': ['ID', 'Status', 'Tierings'],
+                    'additionalProperties': False,
+                    'properties': {
+                        'Id': {'type': 'string'},
+                        'Status': {'enum': ['Enabled', 'Disabled']},
```

Comment:
nit, we've typically used state?

Reply:
The intelligent tiering configuration had `Status` in it which is why I ended up having it. However, I went ahead and changed it to State re: delete for consistency.